packages/atomic/src/components/search/generated-answer/atomic-generated-answer/atomic-generated-answer.tsx

```diff
@@ -275,18 +298,25 @@ export class AtomicGeneratedAnswer implements InitializableComponent {
                     active={this.generatedAnswerState.disliked}
                     onClick={this.clickDislike}
                   />
-                  <CopyButton
-                    title={
-                      !this.copied
-                        ? this.bindings.i18n.t('copy-generated-answer')
-                        : this.bindings.i18n.t('generated-answer-copied')
-                    }
-                    isCopied={this.copied}
-                    onClick={() => {
-                      this.copyToClipboard(this.generatedAnswerState.answer!);
-                      this.generatedAnswer.logCopyToClipboard();
-                    }}
-                  />
+                  {this.showCopyToClipboard && (
+                    <CopyButton
+                      title={this.copyToClipboardTooltip}
+                      isCopied={this.copied}
+                      error={this.copyError}
+                      onClick={async () => {
+                        try {
```

Comment:
Is it possible that this try/catch block is redundant given this.copyToClipboard already has one?
If we add `this.generatedAnswer.logCopyToClipboard();` in there instead I think we can simplify this callback. No?

Reply:
You are absolutely right.